### PR TITLE
Second Save Princess Challenge Complete

### PR DESCRIPTION
**Pull request by Strompy**
# Description
Solves save princess 2 challenge.
Tests for primary and helper functions.
Adds function formatted for copy and pasting into the HackerRank submission box in file `lib/formatted_next_move.rb`

# Closes issue(s)
N/A

# How to test / reproduce
To run test suite from root directory:
`$ ruby test/bot_test.rb`

# Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist
- [X] I have tested this code
- [ ] I have updated the Readme

# Other comments
To submit to HackerRank copy the solution code from lib/formatted_next_move.rb and paste into the submission box

### CODE REVIEW

File: save_princess_2/lib/bot.rb [hunk 1]
```diff
@@ -3,9 +3,31 @@ class Bot
   def initialize; end
 
   def nextMove(n,r,c,grid)
-    puts "0 0"
+    p_r, p_c = find_princess(grid)
+
+    if r < p_r
```
Comment (Strompy) [inline save_princess_2/lib/bot.rb:8]: Could consider refactoring to use a ternary operator if reducing lines was important

File: save_princess_2/lib/bot.rb [hunk 2]
```diff
@@ -17,6 +39,6 @@ def run
         grid[i] = gets
     end
 
-    nextMove(n,r,c,grid)
+    print nextMove(n,r,c,grid)
```
Comment (Strompy) [inline save_princess_2/lib/bot.rb:42]: Standard output for the hackerrank requirements

File: save_princess_2/test/bot_test.rb [hunk 3]
```diff
@@ -1,11 +1,35 @@
...
+  def test_it_exists
+    assert_instance_of Bot, @bot
+  end
+
+  def test_it_can_find_princess
+    grid = ['-----', '-----', 'p--m-', '-----', '-----']
+    assert_equal [2, 0], @bot.find_princess(grid)
+
+    grid2 = ['----p', '-----', '--m--', '-----', '-----']
+    assert_equal [0, 4], @bot.find_princess(grid2)
```
Comment (Strompy) [inline save_princess_2/test/bot_test.rb:19]: Could add edge case test checking for no princess. But that is not likely to be a scenario with the given conditions

File: save_princess_2/test/bot_test.rb [hunk 4]
```diff
@@ -1,11 +1,35 @@
...
+    grid = ['-----', '-----', 'p--m-', '-----', '-----']
+    assert_equal "LEFT", @bot.nextMove(n, r, c, grid)
+
+    grid2 = ['---p', '----', '--m-', '----']
+    n, r, c = 4, 2, 2
+    assert_equal "UP", @bot.nextMove(n, r, c, grid2)
+
+    grid3 = ['-----', '-----', '----p', '-m---', '-----']
+    n, r, c = 5, 3, 1
+    assert_equal "UP", @bot.nextMove(n, r, c, grid3)
```
Comment (Strompy) [inline save_princess_2/test/bot_test.rb:33]: Could add edge case for princess and bot starting at the same point. But that also is not likely with the given conditions

**Comment by Strompy** [comment]
🎉🎉
Passing all hackerrank submissions!
